src/oop/programs_with_interesting_classes/medium/q1/MainTest.java

```diff
@@ -0,0 +1,505 @@
+        };
+        Object car1 = car.createInstance(constructorArguments);
+        Object car2 = car.createInstance(constructorArguments);
+
+        /* Make array copies */
+        String[] partsCopy1 = new String[parts1.length];
+        System.arraycopy(parts1, 0, partsCopy1, 0, parts1.length);
+        String[] partsCopy2 = new String[parts2.length];
+        System.arraycopy(parts2, 0, partsCopy2, 0, parts2.length);
+
+        /* Set fields to values for test */
+        car.setFieldValue(car1, name1, stringAttributeName);
+        car.setFieldValue(car1, speed1, doubleAttributeName);
+        car.setFieldValue(car1, partsCopy1, stringArrayAttributeName);
+        car.setFieldValue(car2, name2, stringAttributeName);
+        car.setFieldValue(car2, speed2, doubleAttributeName);
+        car.setFieldValue(car2, partsCopy2, stringArrayAttributeName);
+
+        /* Call collision */
+        Object[][] collisionInput = {{car1, car.getObjectClass()}, {car2, car.getObjectClass()}};
+        collision.callMethod(collideMethodName, collisionInput);
+
+        /* Test output */
+        if (speed1 == speed2) {
+            String incorrectArray = String.format(
+                    "Your %s method did not correctly modify the %s array when the %s of both %s objects were equal.",
+                    collideMethodName,
+                    stringArrayAttributeName,
+                    doubleAttributeName,
+                    carClassName
+            );
+            _assertArrayEquals(collisionArray(parts1), car.getFieldValue(car1, stringArrayAttributeName), incorrectArray);
+            _assertArrayEquals(collisionArray(parts2), car.getFieldValue(car2, stringArrayAttributeName), incorrectArray);
+        } else {
+            String slowerCar = String.format(
+                    "Your %s method did not correctly modify the %s array of the %s object that had a lower %s.",
+                    collideMethodName,
+                    stringArrayAttributeName,
+                    carClassName,
+                    doubleAttributeName
+            );
+            String fasterCar = String.format(
+                    "Your %s method should not modify the %s array of the %s object that had a higher %s.",
+                    collideMethodName,
+                    stringArrayAttributeName,
+                    carClassName,
+                    doubleAttributeName
+            );
+            if (speed1 > speed2) {
+                _assertArrayEquals(parts1, car.getFieldValue(car1, stringArrayAttributeName), fasterCar);
+                _assertArrayEquals(collisionArray(parts2), car.getFieldValue(car2, stringArrayAttributeName), slowerCar);
+            } else {
+                _assertArrayEquals(collisionArray(parts1), car.getFieldValue(car1, stringArrayAttributeName), slowerCar);
+                _assertArrayEquals(parts2, car.getFieldValue(car2, stringArrayAttributeName), fasterCar);
+            }
+        }
+    }
+
```

Comment:
I'm not sure here but In case we add a copyParts method do we want to test that method also? Or is it better to just test the actual method and do not give them extra grade for that